drivers/imu/iio_adis.c

```diff
@@ -948,38 +957,93 @@ int adis_iio_trigger_handler(struct iio_device_data *dev_data)
 
 	adis = iio_adis->adis_dev;
 
-	ret = adis_read_burst_data(adis, &burst_data);
+	ret = adis_read_burst_data(adis, buff, iio_adis->burst_size);
 	if (ret)
 		return ret;
 
-	if (dev_data->buffer->active_mask & NO_OS_BIT(ADIS_GYRO_X))
-		buff[i++] = adis->burst_sel ? 0 : burst_data.x_gyro;
-	if (dev_data->buffer->active_mask & NO_OS_BIT(ADIS_GYRO_Y))
-		buff[i++] = adis->burst_sel ? 0 : burst_data.y_gyro;
-	if (dev_data->buffer->active_mask & NO_OS_BIT(ADIS_GYRO_Z))
-		buff[i++] = adis->burst_sel ? 0 : burst_data.z_gyro;
-	if (dev_data->buffer->active_mask & NO_OS_BIT(ADIS_ACCEL_X))
-		buff[i++] = adis->burst_sel ? 0 : burst_data.x_accl;
-	if (dev_data->buffer->active_mask & NO_OS_BIT(ADIS_ACCEL_Y))
-		buff[i++] = adis->burst_sel ? 0 : burst_data.y_accl;
-	if (dev_data->buffer->active_mask & NO_OS_BIT(ADIS_ACCEL_Z))
-		buff[i++] = adis->burst_sel ? 0 : burst_data.z_accl;
-	if (dev_data->buffer->active_mask & NO_OS_BIT(ADIS_TEMP))
-		buff[i++] = burst_data.temp_out;
-	if (dev_data->buffer->active_mask & NO_OS_BIT(ADIS_DELTA_ANGL_X))
-		buff[i++] = adis->burst_sel ? burst_data.x_gyro : 0;
-	if (dev_data->buffer->active_mask & NO_OS_BIT(ADIS_DELTA_ANGL_Y))
-		buff[i++] = adis->burst_sel ? burst_data.y_gyro : 0;
-	if (dev_data->buffer->active_mask & NO_OS_BIT(ADIS_DELTA_ANGL_Z))
-		buff[i++] = adis->burst_sel ? burst_data.z_gyro : 0;
-	if (dev_data->buffer->active_mask & NO_OS_BIT(ADIS_DELTA_VEL_X))
-		buff[i++] = adis->burst_sel ? burst_data.x_accl : 0;
-	if (dev_data->buffer->active_mask & NO_OS_BIT(ADIS_DELTA_VEL_Y))
-		buff[i++] = adis->burst_sel ? burst_data.y_accl : 0;
-	if (dev_data->buffer->active_mask & NO_OS_BIT(ADIS_DELTA_VEL_Z))
-		buff[i++] = adis->burst_sel ? burst_data.z_accl : 0;
-
-	return iio_buffer_push_scan(dev_data->buffer, &buff[0]);
+	uint8_t temp_offset = iio_adis->burst_size ? 13 : 7;
+	uint8_t data_cntr_offset = iio_adis->burst_size ? 14 : 8;
+
+	uint16_t current_data_cntr = no_os_get_unaligned_be16((uint8_t*)
```

Comment:
Move these in the variable declaration block, there are several other places where this is the case. Also, using `no_os_bswap_constant_16()` would be cleaner.